dashboard/app/models/stage.rb

```diff
@@ -124,4 +124,30 @@ def summarize
     end
     stage_summary.freeze
   end
+
+  def lockable_state(students)
+    return unless self.lockable?
+
+    # assumption that lockable selfs have a single (assessment) level
```

Comment:
I think when we started this feature there was a bit of a hand-wavy not fully thought out desire that this might be something we wanted to support more broadly, but over time we focused on the feature we knew we needed to support, which is LevelGroup.

If it's okay to have it here for now, I'm okay with taking on a task to see if it might make sense to move some of this logic into LevelGroup in a separate PR.